lib/Compiler.js

```diff
@@ -180,8 +179,12 @@ function compileSolc(embark, contractFiles, contractDirectories, options, callba
           compiledObject[className].gasEstimates = contract.evm.gasEstimates;
           compiledObject[className].functionHashes = contract.evm.methodIdentifiers;
           compiledObject[className].abiDefinition = contract.abi;
+          compiledObject[className].userdoc = contract.userdoc;
           compiledObject[className].filename = filename;
-          compiledObject[className].originalFilename = originalFilename;
+          const normalized = path.normalize(filename);
+          compiledObject[className].originalFilename = path.normalize(contractFiles.find(contractFile => {
```

Comment:
Took me a while to understand what was happening here. For readability's sake, would it be possible to break it up like so?
```suggestion
      const normalized = path.normalize(filename);
      const origContract = contractFiles.find(contractFile => normalized.includes(path.normalize(contractFile.originalPath)));
      if(origContract) compiledObject[className].originalFilename = path.normalize(origContract.originalPath);
```
Also adds some error handling in the oddball case the contract can't be found...